subprojects/core/src/main/java/org/gradle/api/internal/file/copy/CopySpecInternal.java

```diff
@@ -53,6 +54,9 @@ public interface CopySpecInternal extends SyncSpec {
 
     CopySpecInternal preserve(Action<? super PatternFilterable> action);
 
+    @Nullable
+    File getDestinationDir();
```

Comment:
Already available as destinationDir or destPath, this makes retrieval of the output path consistent.